### PR TITLE
Replace conditional data dump by a triage function in choosetests.jl

### PR DESCRIPTION
**Pull request by o314**
*Following #35209 (fork unavailable from now)*

This patch implements the comment at https://github.com/JuliaLang/julia/pull/35209#discussion_r407955751

### CODE REVIEW

File: test/choosetests.jl [hunk 1]
```diff
@@ -82,59 +82,27 @@ function choosetests(choices = [])
...
-
-    stringtests = ["strings/basic", "strings/search", "strings/util",
-                   "strings/io", "strings/types"]
-    if "strings" in skip_tests
-        filter!(x -> (x != "strings" && !(x in stringtests)), tests)
-    elseif "strings" in tests
-        # specifically selected case
-        filter!(x -> x != "strings", tests)
-        prepend!(tests, stringtests)
+    function patchlib!(tests, name, files=[name])
```
Comment (StefanKarpinski) [inline test/choosetests.jl:85]: Calling this `patchlib!` is a little weird. That makes me think of patchelf and other such library patching utils.

Comment (o314) [reply test/choosetests.jl:85]: 1. probably not the best name one can choose. first choice was the last one o:

2. functional prg @ code leads to delta-oriented programming @ data. 
    diff, patch etc can spread and disseminate. that's great!

3. no problem at all to change this name. 
    i will try to get inspiration for that. hum, hum.

**Comment by o314** [comment]
a few couple of work outdoor and i will go back to fix that.

File: test/choosetests.jl (hunk 1 above)
Comment (rfourquet) [reply test/choosetests.jl]: The new name `filtertests!` is indeed more descriptive!

**Comment by rfourquet** [comment]
Thanks!

**Pull request merged**